geode/src/main/java/org/apache/calcite/adapter/geode/rel/GeodeFilter.java

```diff
@@ -233,6 +238,90 @@ private String translateOp2(String op, String name, RexLiteral right) {
       }
       return name + " " + op + " " + valueString;
     }
+
+    /**
+     *  Get the field name for the left node to use for IN SET query
+     */
+    private String getLeftNodeFieldName(RexNode left) {
+      switch (left.getKind()) {
+      case INPUT_REF:
+        final RexInputRef left1 = (RexInputRef) left;
+        return fieldNames.get(left1.getIndex());
+      case CAST:
+        // FIXME This will not work in all cases (for example, we ignore string encoding)
+        return getLeftNodeFieldName(((RexCall) left).operands.get(0));
+      case OTHER_FUNCTION:
+        return left.accept(new GeodeRules.RexToGeodeTranslator(this.fieldNames));
+      default:
+        return null;
+      }
+    }
+
+    /**
+     *  Check if we can use IN SET Query clause to improve query performance
+     */
+    private boolean useInSetQueryClause(List<RexNode> disjunctions) {
+      Set<String> leftFieldNameSet = new HashSet<>();
```

Comment:
Is it possible to avoid "global" variable (`leftFieldNameSet`) ? It seems like you're using "state" in lambda. Is it really necessary ?